shell/platform/windows/accessibility_root_node.cc

```diff
@@ -0,0 +1,271 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/windows/accessibility_root_node.h"
+
+#include "flutter/fml/logging.h"
+
+namespace flutter {
+
+static constexpr LONG kWindowChildId = 1;
+static constexpr LONG kInvalidChildId = 3;
+
+AccessibilityRootNode::AccessibilityRootNode() : alert_accessible_(nullptr) {}
+
+AccessibilityRootNode::~AccessibilityRootNode() {
+  if (alert_accessible_) {
+    alert_accessible_->Release();
+    alert_accessible_ = nullptr;
+  }
+}
+
+IAccessible* AccessibilityRootNode::GetTargetAndChildID(VARIANT* var_id) {
+  LONG& child_id = var_id->lVal;
+  if (V_VT(var_id) != VT_I4) {
+    child_id = kInvalidChildId;
+    return FALSE;
+  }
+  child_id = V_I4(var_id);
+  if (!window_accessible_) {
+    return nullptr;
+  }
+  if (child_id == CHILDID_SELF || child_id == kWindowChildId) {
+    child_id = CHILDID_SELF;
```

Comment:
What's the value of setting `child_id` on this line and line 38 below?

Reply:
`child_id` is a reference to `var_id->lVal`. Setting it updates the value in the `VARIANT` passed to the function, so for example, if a method is called in the root node with `var_id.lVal == 1` pointing to the window node child, `var_id.lVal` is set to `CHILDID_SELF = 0` and the method is called on the window node.

Reply:
Ah I missed the reference, thanks!

Reply:
I should make stronger coffee 😂

Reply:
I wonder if this method would be clearer if it returned the target and child ID instead of mutating the input `var_id`. Perhaps it could return a `std::optional<std::pair<IAccessible*, VARIANT>>`? This would be more verbose but also make its behavior more apparent. Thoughts?